src/textual/widget.py

```diff
@@ -1040,7 +1040,9 @@ def refresh(
 
         if layout:
             self._layout_required = True
-            self._clear_arrangement_cache()
+            if isinstance(self._parent, Widget):
```

Comment:
When a widget's dimensions / visibility change, we want to clear the *parent's* arrangement cache.